### PR TITLE
Classes overview in Teachers Page

### LINKED ISSUE #685: Change text into actual link
There might be a reason for this but it is better if this link is a real link so you can copy-paste with right-click too (in join class)

image

### LINKED ISSUE #702: Add e2e test for link to latest_shared program in class overview
Enhance the e2e tests to check the integrity of the shared program link in the class overview. This would have prevented the bug fixed in #696

### PR DESCRIPTION
**Pull request by fpereiro**
- Move classes overview to For Teachers page.
- Hide For Teachers page from main menu unless the user is a teacher.
- Related to (but doesn't close) #546 
- In profile, hides "My classes" for teachers, and puts that section on top of the profile for all other users.
- Improves E2E tests for classes, closes #702 
- Expandable/shrinkable docs (WIP, see below).
- Use syntax highlighting for code snippets in For Teachers page.
- Use a markdown -> HTML pipeline for parsing the markdown docs.
- Load auth messages through the same mechanism that UI & error messages are loaded (`client_messages.js`), to avoid empty modal messages and save load time.
- In class overview, shows link as an actual link (closes #685)
- If a user tries to join a class that they're part of, the page will let them know they're already part of the class.
- Adds rename button to class overview.
- Allows teachers to see list of all programs for students that are part of their classes.
- Note at the top of teachers' profiles that states that the list of their classes has been moved.

**How to test**:
- For Teachers page should not be visible in the heading of either a non-logged in user or a non-teacher user, but it should be visible for users that are teachers.
- If non-teachers go to /for-teachers.html, they can still see the docs but not the table of classes.
- In Profile, teacher users should not be able to see the My Classes section.
- The top of the For Teachers page should contain a table with all the classes owned by the teacher.
- The bottom of the For Teachers page should have a set of docs with sections that can be expanded/shrinked. The code snippets in these docs should be highlighted properly.
- In class overview, shows link as an actual link
- If a user tries to join a class that they're part of, the page will let them know they're already part of the class.
- Rename a class as teacher.
- Be able to see list of programs for each student as a teacher.
- See note at the top of teachers' profile stating that classes have moved.

### CODE REVIEW

**Comment by Felienne** [comment]
Clicking a class gives an error (maybe because it is empty?) 

 image

That means I cannot test the rest of the flow cause I can't invite new kids into the class

**Comment by Felienne** [comment]
Unified modal (nice that we now use that everywhere!) is empty for Dutch:

image

**Comment by Felienne** [comment]
Hi @fpereiro I tried the normal flow (make class, follow join lonk from a different account) but I run into several new and weird issues: stopping pages, logging out all of a sudden and more weird thing. Hard to pinpoint this but I guess a first step is you do a few more run throughs to see if it is all in order for you.

**Comment by Felienne** [comment]
While I am trying to figure out all the weird issues (not yet succeeded in that), there are some cosmetic ones too. "Ja" is a bit weird here, "ok" would be better:

image

File: build-tools/validate-e2e [hunk 1]
```diff
@@ -10,7 +10,7 @@ pids=""
 trap 'kill $pids' EXIT
 
 export PORT=5050
-env HEROKU_APP_NAME=localhost SECRET_KEY=TheSecret python app.py > e2e-server.log 2>&1 &
+env HEROKU_APP_NAME=localhost BASE_URL=http://localhost:$PORT SECRET_KEY=TheSecret python app.py > e2e-server.log 2>&1 &
```
Comment (fpereiro) [inline build-tools/validate-e2e:13]: Hi @rix0rrr ! I just added the `BASE_URL` env variable to the script that runs the E2E tests. We need that variable now on a couple places other than the email sending. Let me know if this looks right to you as well.

**Review by Felienne — CHANGES_REQUESTED** [review]
Hi @fpereiro I just tested it again and it seems to be fine now!

I have two tiny remarks:
* Can we place a note in profile (just a line at the top), for logged in teachers, telling them the class overview is moved?
* If a kid tries to join a class they are already in, can we give them a warning? Rather than rendering the page and making hem click again, can we give them the warning when they clink the first link (only when that is easy to make, don't spend lots of time on it!)

File: templates/profile.html [hunk 2]
```diff
@@ -69,13 +69,6 @@ <h2>{{auth.student_classes}}</h2>
     <ul class="list-none" style="margin: 0"></ul>
     <br><br>
   </div>
-  <div id="teacher_classes" class="hidden pl-8">
```
[the remark this answers is not in the document]
Comment (Felienne) [reply templates/profile.html:72]: This is fixed!

**Comment by fpereiro** [comment]
Summary of missing work:
- Add a list of shared programs for each student in a column.
- Add a note that tells teachers that Class Overview moved.
- Change text into actual link (closes #685)
- Add rename class button
- When joining a class a user already joined, tell them that they have already joined the class.

**Comment by fpereiro** [comment]
@Felienne ready for review again!

**Pull request merged**